tests/unit/DebugUtilsTest.ts

```diff
@@ -648,4 +653,691 @@ describe('DebugUtils', () => {
+                    policyID: '1',
+                    managerID: 12345,
+                    type: CONST.REPORT.TYPE.EXPENSE,
+                    stateNum: CONST.REPORT.STATE_NUM.SUBMITTED,
+                    statusNum: CONST.REPORT.STATUS_NUM.SUBMITTED,
+                },
+            };
+            const MOCK_REPORT_ACTIONS: ReportActions = {
+                // eslint-disable-next-line @typescript-eslint/naming-convention
+                '0': {
+                    reportActionID: '0',
+                    actionName: CONST.REPORT.ACTIONS.TYPE.CREATED,
+                    created: '2024-08-08 18:70:44.171',
+                },
+                // eslint-disable-next-line @typescript-eslint/naming-convention
+                '1': {
+                    reportActionID: '1',
+                    actionName: CONST.REPORT.ACTIONS.TYPE.REPORT_PREVIEW,
+                    created: '2024-08-08 19:70:44.171',
+                    childReportID: '2',
+                },
+            };
+            await Onyx.multiSet({
+                ...MOCK_REPORTS,
+                [`${ONYXKEYS.COLLECTION.REPORT_NAME_VALUE_PAIRS}2` as const]: {
+                    // eslint-disable-next-line @typescript-eslint/naming-convention
+                    private_isArchived: false,
+                },
+                [`${ONYXKEYS.COLLECTION.REPORT_ACTIONS}1` as const]: MOCK_REPORT_ACTIONS,
+                [`${ONYXKEYS.COLLECTION.POLICY}1` as const]: {
+                    approvalMode: CONST.POLICY.APPROVAL_MODE.BASIC,
+                    type: CONST.POLICY.TYPE.CORPORATE,
+                },
+                [ONYXKEYS.SESSION]: {
+                    accountID: 12345,
+                },
+            });
+            // eslint-disable-next-line @typescript-eslint/non-nullable-type-assertion-style
+            const {reportAction} = DebugUtils.getReasonAndReportActionForGBRInLHNRow(MOCK_REPORTS[`${ONYXKEYS.COLLECTION.REPORT}1`] as Report) ?? {};
+            expect(reportAction).toMatchObject(MOCK_REPORT_ACTIONS['1']);
+        });
+        it('returns undefined report action when report has no GBR', () => {
+            const {reportAction} =
+                DebugUtils.getReasonAndReportActionForGBRInLHNRow({
+                    reportID: '1',
+                }) ?? {};
+            expect(reportAction).toBeUndefined();
+        });
+    });
+    describe('getRBRReportAction', () => {
+        beforeAll(() => {
+            Onyx.init({
+                keys: ONYXKEYS,
+            });
+        });
+        beforeEach(() => {
+            Onyx.clear();
+        });
+        it('returns undefined when report has no RBR', () => {
+            const reportAction = DebugUtils.getRBRReportAction(
+                {
+                    reportID: '1',
+                },
+                undefined,
+            );
+            expect(reportAction).toBeUndefined();
+        });
+        // TODO: remove '.failing' once the implementation is fixed
```

Comment:
NAB: Do we have an issue to track this?

Reply:
Would you please create one or ask in Slack to have an issue created?

Reply:
@neil-marcellini I just checked and it seems there's no implementation issue.

The test is just outdated with the new implementation, so I'll fix it in this PR - https://github.com/Expensify/App/pull/50468- if that's okay with you? :D